Test with Symfony 8

loic425 opened the pull request:
blocked by https://github.com/phpcr/phpcr-utils/pull/227

loic425 committed:
Add support for Symfony 8

dbu commented:
thanks a lot for looking into this. i updated phpstan, please rebase on 2.x. once i tagged phpcr-utils, i will run tests here and see if there is anything left to do.

loic425 commented:
I've rebased the PR and renamed it cause there are no main dependencies to Symfony. It's only used on dev requirements as we do in this PR https://github.com/doctrine/persistence/pull/477.

dbu commented on .github/workflows/test-application.yaml:
```diff
@@ -35,12 +35,19 @@ jobs:
             db: sqlite
           - php-version: '8.0'
             dependencies: lowest
+            symfony-version: '7.*'
```
```suggestion
            symfony-version: '6.*'
```

dbu committed:
remove deprecated xml_parser_free that was a noop since php 8.0

dbu committed:
only call setAccessible in PHP 8.0 as afterwards it is no longer needed

dbu committed:
use pdo\sqlite::createFunction if available

dbu commented:
thanks a lot for the contribution!
